utils/src/config.rs

```diff
@@ -209,32 +213,11 @@ pub const fn default_cmd_workers() -> u8 {
     8
 }
 
-impl CurpConfig {
-    /// Create a new server timeout
-    #[must_use]
-    #[inline]
-    #[allow(clippy::too_many_arguments)]
-    pub fn new(
```

Comment:
Because the config is usually immutable and all its field is pub, we can create a struct without the construction function. I also think maybe it's a better practice to eliminate all the construction functions of our config struct. As an example, `CurpConfig { heartbeat_intervals: 0.3s, cmd_workers: 8, ...}` is more understandable than `CurpConfig::new(0.3s, 8, ...)`. We are writing Rust not Java🤣.